duecredit/collector.py

```diff
@@ -219,25 +222,21 @@ def cite(self, entry, **kwargs):
         # TODO: if cite is invoked but no path is provided -- we must figure it out
         # I guess from traceback, otherwise how would we know later to associate it
         # with modules???
+        path = kwargs.get('path', None)
+        if path is None:
+            raise ValueError('path must be provided')
+
         if isinstance(entry, DueCreditEntry):
             # new one -- add it
             self.add(entry)
-            entry_ = entry
+            entry_ = self._entries[entry.get_key()]
         else:
             entry_ = self._entries[entry]
         entry_key = entry_.get_key()
 
-        # TODO: we must allow the same entry be present in multiple citations, so
-        # RF to do so
-        if entry_key not in self.citations:
-            self.citations[entry_key] = Citation(entry_, **kwargs)
-
-        citation = self.citations[entry_key]
-        citation.count += 1
-        # TODO: update path and use here?
-        if not citation.path:
-            citation.path = kwargs.get('path', None)
-
+        citation = Citation(entry_, **kwargs)
+        # update entry count
+        entry_.count += 1
```

Comment:
My thinking was that we want to count how many times a single reference has been cited, regardless of where it was used. But apparently this is not what you thought of :-) but then, what is the advantage of dividing the count at the level of path? to get how many times a single object has been called?